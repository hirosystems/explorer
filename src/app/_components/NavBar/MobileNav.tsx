import { useColorMode } from '@chakra-ui/react';
import React, { FC } from 'react';
import { PiX } from 'react-icons/pi';
import { TbX } from 'react-icons/tb';

import { Flex } from '../../../ui/Flex';
import { IconButton } from '../../../ui/IconButton';
import { Stack } from '../../../ui/Stack';
import { ColorModeButton } from './ColorModeButton';
import { MobileNavItem } from './MobileNavItem';
import { NavItem } from './types';

export const MobileNav: FC<{ navItems: NavItem[]; close: () => void }> = ({ navItems, close }) => {
  const colorMode = useColorMode().colorMode;
  return (
    <Stack
      position="fixed"
      height="full"
      width="full"
      backgroundColor="bg"
      top={0}
      left={0}
      zIndex={2}
      padding={6}
      gap={6}
    >
      <Flex justifyContent={'space-between'}>
        <ColorModeButton aria-label={'Change color mode'} color="invert" borderWidth={'1px'} />
        <IconButton onClick={close} icon={<PiX />} aria-label={'Close menu'} />
      </Flex>
      {navItems.map(navItem => (
        <MobileNavItem key={navItem.id} {...navItem} />
      ))}
    </Stack>
  );
};
