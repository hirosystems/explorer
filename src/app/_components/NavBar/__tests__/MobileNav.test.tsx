import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import React from 'react';

import { MobileNav } from '../MobileNav';

describe('MobileNav', () => {
  it('should render correctly', () => {
    const mockNavItems = [
      {
        id: '1',
        label: 'Mobile Sample Item 1',
        href: '#',
        children: [
          {
            id: '1.1',
            label: 'Mobile Sub Item 1',
            href: '#',
          },
        ],
      },
      {
        id: '2',
        label: 'Mobile Sample Item 2',
        href: '#',
      },
    ];

    const mockClose = jest.fn();

    const { asFragment } = render(<MobileNav navItems={mockNavItems} close={mockClose} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
