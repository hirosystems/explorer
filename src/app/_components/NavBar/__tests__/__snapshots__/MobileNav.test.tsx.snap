// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNav should render correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-1kyhbj8"
  >
    <div
      class="css-gg4vpm"
    >
      <button
        aria-label="Change color mode"
        class="chakra-button css-1fthbo8"
        title="Toggle color mode"
        type="button"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M233.54,142.23a8,8,0,0,0-8-2,88.08,88.08,0,0,1-109.8-109.8,8,8,0,0,0-10-10,104.84,104.84,0,0,0-52.91,37A104,104,0,0,0,136,224a103.09,103.09,0,0,0,62.52-20.88,104.84,104.84,0,0,0,37-52.91A8,8,0,0,0,233.54,142.23ZM188.9,190.34A88,88,0,0,1,65.66,67.11a89,89,0,0,1,31.4-26A106,106,0,0,0,96,56,104.11,104.11,0,0,0,200,160a106,106,0,0,0,14.92-1.06A89,89,0,0,1,188.9,190.34Z"
          />
        </svg>
      </button>
      <button
        aria-label="Close menu"
        class="chakra-button css-12w4gwb"
        type="button"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
          />
        </svg>
      </button>
    </div>
    <div
      class="chakra-stack css-1q7sb2n"
    >
      <a
        class="css-tz7dko"
        href="#"
      >
        <span
          class="chakra-text css-13o7eu2"
        >
          Mobile Sample Item 1
        </span>
        <svg
          class="chakra-icon css-3tv48m"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
          />
        </svg>
      </a>
      <div
        class="chakra-collapse"
        style="overflow: hidden; display: none; opacity: 0; height: 0px;"
      >
        <div
          class="chakra-stack css-1s2q0he"
        >
          <a
            class="css-1ak55br"
            href="#"
          >
            Mobile Sub Item 1
          </a>
        </div>
      </div>
    </div>
    <div
      class="chakra-stack css-1q7sb2n"
    >
      <a
        class="css-tz7dko"
        href="#"
      >
        <span
          class="chakra-text css-13o7eu2"
        >
          Mobile Sample Item 2
        </span>
      </a>
      <div
        class="chakra-collapse"
        style="overflow: hidden; display: none; opacity: 0; height: 0px;"
      >
        <div
          class="chakra-stack css-1s2q0he"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
